src/hazmat/sieve.rs

```diff
@@ -11,21 +11,46 @@ use rand_core::CryptoRngCore;
 use crate::hazmat::precomputed::{SmallPrime, LAST_SMALL_PRIME, RECIPROCALS, SMALL_PRIMES};
 use crate::traits::SieveFactory;
 
-/// Returns a random odd integer with given bit length
+/// Set bits in the returned number.
+#[derive(Debug, Clone, Copy)]
+pub enum SetBits {
+    /// No additional bits set.
+    None,
+    /// Set the most significant bit.
+    Msb,
+    /// Set two most significant bits.
+    ///
+    /// This is useful in the RSA case because a product of two such numbers will have a guaranteed bit size.
+    TwoMsb,
+}
```

Comment:
```suggestion
pub enum SetBits {
    /// Set the most significant bit, thus limiting the range to `[MAX/2+1, MAX]`. The default.
    Msb,
    /// Set two most significant bits, limiting the range to `[MAX - MAX/4 + 1, MAX]`.
    ///
    /// This is useful in the RSA case because a product of two such numbers will have a guaranteed bit size.
    TwoMsb,
    /// No additional bits set; uses the full range `[0, MAX]`.
    None,
}
```

I think the default is best kept first.